Utilities.java

```diff
@@ -1,12 +1,33 @@
-import java.io.*;
 import java.util.*;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.*;
 
 enum LetterComparisonResult
 {
-	UNKNOWN,
-	EXACT,
-	WRONG,
-	WRONG_INDEX,
+	UNKNOWN("x"),
+	EXACT("#"),
+	WRONG("."),
+	WRONG_INDEX("?");
+
+	private String stringRepresentation;
+
+	LetterComparisonResult(String stringRepresentation)
+	{
+		this.stringRepresentation = stringRepresentation;
+	}
+
+	String getStringRepresentation()
+	{
+		return stringRepresentation;
+	}
+
+	public static Optional<LetterComparisonResult> fromStringRepresentation(String stringRepresentation)
```

Comment:
What is this string parameter? Is it a string representation of a full word's results or the string representation of a single result value? If the latter, is it possible to assign chars to the enum values instead of strings? Would it be clearer if all these "string representation" names were "char representation"?

Reply:
The `stringRepresentation` param is a String of a single character (`.`, `#`, `?`)

I can definitely assign chars instead of strings and rename the field